synapse/http/matrixfederationclient.py

```diff
@@ -165,10 +162,7 @@ def __attrs_post_init__(self) -> None:
 
         destination_bytes = self.destination.encode("ascii")
         path_bytes = self.path.encode("ascii")
-        if self.query:
-            query_bytes = encode_query_args(self.query)
-        else:
-            query_bytes = b""
```

Comment:
I hadn't thought of this, but fortunately it will work:

```
$ python
Python 3.10.4 (main, Mar 25 2022, 00:00:00) [GCC 11.2.1 20220127 (Red Hat 11.2.1-9)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib.parse
>>> urllib.parse.urlencode({}, True)
""
```